schema-engine-tag/src/main/java/org/apache/iotdb/db/metadata/tagSchemaRegion/tagIndex/query/MemTableQuery.java

```diff
@@ -59,10 +61,12 @@ public List<MemChunkGroup> getChildren(
    * @param context query request context
    */
   @Override
-  public void query(MemTable memNode, QueryRequest queryRequest, QueryRequestContext context) {
+  public void query(
+      MemTable memNode, SingleQueryRequest queryRequest, QueryRequestContext context) {
     // if the memTable is immutable, we need to delete the id in deletionList in the query result
     if (memNode.isImmutable()) {
-      RoaringBitmap roaringBitmap = context.getValue();
+      RoaringBitmap roaringBitmap =
+          (RoaringBitmap) Optional.ofNullable(context.getValue()).orElse(new RoaringBitmap());
```

Comment:
It seems there's raw usage of generic RequestContext, the response stored in which can be explicit.